paraatm/bdn/_bdn.py

```diff
@@ -0,0 +1,157 @@
+"""
+Bayesian deep neural network (BDN) class
+---
+Required pakages:
+keras
+numpy
+
+"""
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Input, Dense, Dropout,LSTM,Activation
+from keras.optimizers import Adam
+from keras.regularizers import l1,l2
+from keras.models import Sequential
+import numpy as np
+
+class BDN(object):
+    """
+    BDN class creates bayesian deep neural network and provides 
+    predictions with uncertainty
+    
+    Reference: Gal, Y., & Ghahramani, Z. (2016, June). Dropout as a bayesian approximation: Representing model uncertainty in deep learning. In international conference on machine learning (pp. 1050-1059).
+    
+    Attributes
+    ----------
+    x_train: training data
+    y_train: training target
+    x_test: test data
+    y_test: test target
+    idrop: dropout rate for input layer
+    odrop: dropout rate for output layer
+    rdrop:dropout rate for the recurrent layer(DNN model doesn't need rdrop
+    but is provided by default)
+    weight_decay: regularization factor
+    lr: learning rate
+    num_unit: number of unit at each layer
+    batch_size: mini batch size
+    epochs: number of epochs
+    iter_: number of predictions for each sample
+    results: array of predicted results for the test data.
+    
+    Methods:
+    -------
+    get_RNN_model
+    get_DNN_model
+    BDNpred
+    
+    """
+    def __init__(self,x_train,y_train,x_test,y_test,idrop=0.,
+                 odrop=0.25,rdrop=0.25,
+                 weight_decay=1e-4,lr=1e-3,num_unit=100,
+                 batch_size=30,epochs=200,iter_=5):
+        self.x_train = x_train
+        self.y_train = y_train
+        self.x_test = x_test
+        self.y_test = y_test
+        self.idrop = idrop
+        self.odrop = odrop
+        self.rdrop = rdrop
+        self.weight_decay = weight_decay
+        self.lr = lr
+        self.num_unit=num_unit
+        self.batch_size = batch_size
+        self.epochs = epochs
+        self.iter_ = iter_
+        if not isinstance(self.x_train,np.ndarray):
+            raise Exception('Wrong type: expect an array!') 
+
+    def get_RNN_model(self):
```

Comment:
Some of the methods of the `BDN` class may be for internal use as opposed to being part of the "public API" that users would access.  It is helpful to indicate this by prefixing all of the "private" methods with an underscore, e.g., `_get_RNN_model`.  This helps users know what functions they may want to use, and it also used by documentation systems as well.  I am not sure exactly which of your methods are intended to be public, but it looks like `BDNpred` may be the only one.